mentat/parsers/file_edit.py

```diff
@@ -0,0 +1,196 @@
+            self.rename_file_path = None
+        return True
+
+    def filter_replacements(
+        self,
+        code_file_manager: CodeFileManager,
+        user_input_manager: UserInputManager,
+        config: ConfigManager,
+    ) -> bool:
+        if self.is_creation:
+            display_information = DisplayInformation(
+                self.file_path, [], [], [], FileActionType.CreateFile, None, None, None
+            )
+            if not _ask_user_change(
+                user_input_manager, display_information, "Create this file?"
+            ):
+                return False
+            file_lines = []
+        else:
+            rel_path = Path(os.path.relpath(self.file_path, config.git_root))
+            file_lines = code_file_manager.file_lines[rel_path]
+
+        if self.is_deletion:
+            display_information = DisplayInformation(
+                self.file_path, [], [], file_lines, FileActionType.DeleteFile
+            )
+            if not _ask_user_change(
+                user_input_manager, display_information, "Delete this file?"
+            ):
+                return False
+
+        if self.rename_file_path is not None:
+            display_information = DisplayInformation(
+                self.file_path,
+                [],
+                [],
+                [],
+                FileActionType.RenameFile,
+                new_name=self.rename_file_path,
+            )
+            if not _ask_user_change(
+                user_input_manager, display_information, "Rename this file?"
+            ):
+                self.rename_file_path = None
+
+        new_replacements = list[Replacement]()
+        for replacement in self.replacements:
+            removed_block = file_lines[
+                replacement.starting_line : replacement.ending_line
+            ]
+            display_information = DisplayInformation(
+                self.file_path,
+                file_lines,
+                replacement.new_lines,
+                removed_block,
+                FileActionType.UpdateFile,
+                replacement.starting_line,
+                replacement.ending_line,
+                self.rename_file_path,
+            )
+            if _ask_user_change(
+                user_input_manager, display_information, "Keep this change?"
+            ):
+                new_replacements.append(replacement)
+        self.replacements = new_replacements
+
+        return (
+            self.is_creation
+            or self.is_deletion
+            or (self.rename_file_path is not None)
+            or len(self.replacements) > 0
+        )
+
+    def resolve_conflicts(self, user_input_manager: UserInputManager):
+        self.replacements.sort(reverse=True)
+        for index, replacement in enumerate(self.replacements):
+            for other in self.replacements[index + 1 :]:
+                # TODO: another type of conflict (not caught here) would be both replacements being inserts on same line
```

Comment:
so what happens in that scenario now? does Mentat crash? or does it just randomly put one first?

Reply:
Right now it just puts the insertion first in the list first and the next insertion last

Reply:
how about we treat that scenario the same as the other conflict scenario then, and alert / display to the user what's happening